quic/s2n-quic-core/src/recovery/bbr/windowed_filter.rs

```diff
@@ -193,4 +270,52 @@ mod tests {
         assert_eq!(Some(1), filter.value());
         assert_eq!(Some(now), filter.last_updated);
     }
+
+    #[test]
+    fn min_rtt_filter() {
+        let mut filter = MinRttWindowedFilter::new();
+
+        // Filter has not received an update, so no value should be present
+        assert_eq!(None, filter.min_rtt());
+        // Probe RTT is not expired upon initialization
+        assert!(!filter.probe_rtt_expired());
+
+        // After the first update, the first value is the min
+        let now = NoopClock.get_time();
+        filter.update(Duration::from_millis(5), now);
+        assert_eq!(Some(Duration::from_millis(5)), filter.min_rtt());
+        assert!(!filter.probe_rtt_expired());
+
+        // PROBE_RTT_INTERVAL has passed
+        let now = now + PROBE_RTT_INTERVAL;
+        filter.update(Duration::from_millis(9), now);
+        // min_rtt is not updated since it has not expired and is lower than the new rtt
+        assert_eq!(Some(Duration::from_millis(5)), filter.min_rtt());
+        assert!(filter.probe_rtt_expired());
+
+        // Midway through the Probe RTT period the RTT drops
+        let now = now + Duration::from_secs(2);
+        let probe_rtt_update_time = now;
+        filter.update(Duration::from_millis(7), now);
+        // min_rtt is not updated since it has not expired and is lower than the new rtt
+        assert_eq!(Some(Duration::from_millis(5)), filter.min_rtt());
+        assert!(!filter.probe_rtt_expired());
+        assert_eq!(Some(Duration::from_millis(7)), filter.min_probe_rtt.value());
+
+        // Now the Min RTT has expired
+        let now = now + Duration::from_secs(3);
+        filter.update(Duration::from_millis(10), now);
+        // min_rtt is updated since it has expired. The value is set to the current probe_rtt
+        assert_eq!(Some(Duration::from_millis(7)), filter.min_rtt());
+        // min_rtt last_updated is set to the probe_rtt last updated time
+        assert_eq!(Some(probe_rtt_update_time), filter.min_rtt.last_updated);
+        assert!(!filter.probe_rtt_expired());
```

Comment:
can you explain what exactly causes the min RTT to expire. Is it `Duration::(10)` and if so can you use `MIN_RTT_FILTER_LEN` here instead

Reply:
its because the `MIN_RTT_FILTER_LEN` has passed since the first time the min_rtt was set (on line 285). First the PROBE_RTT_LEN passes (5 seconds), then 2 seconds, and finally 3 seconds. Since  5 + 2 + 3 = 10 = MIN_RTT_FILTER_LEN, the min_rtt expires. I'll make the comment a little more clear